src/reconcile/rec_write.c

```diff
@@ -3680,25 +3655,26 @@ __wt_bulk_insert_row(WT_SESSION_IMPL *session, WT_CURSOR_BULK *cbulk)
 	    cursor->value.data, cursor->value.size, (uint64_t)0));
 
 	/* Boundary: split or write the page. */
-	if (key->len + val->len > r->space_avail) {
-		if (r->raw_compression)
-			WT_RET(
-			    __rec_split_raw(session, r, key->len + val->len));
-		else {
-			/*
-			 * Turn off prefix compression until a full key written
-			 * to the new page, and (unless already working with an
-			 * overflow key), rebuild the key without compression.
-			 */
-			if (r->key_pfx_compress_conf) {
-				r->key_pfx_compress = false;
-				if (!ovfl_key)
-					WT_RET(__rec_cell_build_leaf_key(
-					    session, r, NULL, 0, &ovfl_key));
-			}
+	if (r->raw_compression) {
+		if (key->len + val->len > r->space_avail)
+			WT_RET(__rec_split_raw(
+			    session, r, key->len + val->len));
+	} else if (WT_CHECK_CROSSING_BND(r, key->len + val->len)) {
```

Comment:
That is what I did in the earlier revisions. I was getting perf degradation in the highly optimised bulk insert case. I fixed some amount of that degradation, but couldn't get rid of all of it. Most of it came from memmove of the second chunk to the start of interim buffer.

Also, since bulk load creates pages of page_size (split_pct as 100), it doesn't make sense to do all this change for this path (like it doesn't make sense for fixed size column store).
So I am bypassing my change for the bulk load routine.